packages/terra-folder-tree/src/subcomponents/FolderTreeItem.jsx

```diff
@@ -68,22 +74,27 @@ const FolderTreeItem = ({
   onClick,
   onToggle,
   subfolderItems,
+  parentRef,
   intl,
 }) => {
   const theme = useContext(ThemeContext);
   const isFolder = subfolderItems?.length > 0;
+  const itemNode = useRef();
+  const subFolderNode = useRef();
 
   const subfolder = isFolder ? (
```

Comment:
It seems like we should have separate React components for the Folder vs the Items.  Maybe it is a future story, but I think the logic would be cleaner with that architecture.